### DISCUSSION TITLE
How to bind to Task<T> itself?

### DISCUSSION THREAD

**Discussion by segln** [AvaloniaUI/Avalonia#13753] [Q&A] [answered]
I used `^` with `.`(Task itself) but not works.
It just binds to Task<string> object.

### Sample code
```C#
public class ViewModel : ViewModelBase
{
    public ObservableCollection<Task<string>> Test { get; } = [Task.FromResult("fail")];
}
```
```xaml
<ComboBox ItemsSource="{CompiledBinding Test}">
    <ComboBox.ItemTemplate>
        <DataTemplate>
            <DataTemplate.DataType>
                <x:Type TypeName="tasks:Task" x:TypeArguments="system:String"/>
            </DataTemplate.DataType>
            <Button Content="{CompiledBinding .^}"></Button>
        </DataTemplate>
    </ComboBox.ItemTemplate>
</ComboBox>
```
### Result
![image](https://github.com/AvaloniaUI/Avalonia/assets/18192378/12b80ebc-62ca-4fdb-bfda-76c35de3c517)

**Comment by timunie** [comment]
Not sure if that is going to work. Also please note, that an ObesvableCollection should not be created or populated form a Thread other than the UI thread. If you need lazy loading, either fetch the data into an array first and after all data is loaded, add the items to your collection or check out Dynamic Data which is shipped with Reactive UI.

**Reply by segln** [reply]
Then there is no way to bind to Task with `Path=.`?

**Reply by timunie** [reply, marked as answer]
Looks like it is not. Maybe need to wrap it in an ItemViewModel.

**Reply by segln** [reply]
Thanks! Is it by-designed or will it become possible in the future?
